src/v/v8_engine/tests/script_dispatcher_test.cc

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "seastarx.h"
+#include "utils/file_io.h"
+#include "v8_engine/executor.h"
+#include "v8_engine/script_dispatcher.h"
+
+#include <seastar/core/reactor.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/algorithm/string/case_conv.hpp>
+#include <boost/test/tools/old/interface.hpp>
+
+class executor_wrapper_for_test {
+public:
+    executor_wrapper_for_test()
+      : _executor(ss::engine().alien(), 1, ss::smp::count) {}
+
+    ~executor_wrapper_for_test() { _executor.stop().get(); }
+
+    v8_engine::executor& get_executor() { return _executor; }
+
+private:
+    v8_engine::executor _executor;
+};
+
+v8_engine::enviroment env;
+
+SEASTAR_THREAD_TEST_CASE(simple_update_test) {
```

Comment:
I would add more complicated tests that test the internal mutex, or whatever synchronization mechanism you decided to do. So if there is a configurable limit, make a test to exceed the limit

Reply:
Yep, on this iteration I wanna get some feedback for script_dispather